internal/mithril-build-script/src/open_api.rs

```diff
@@ -0,0 +1,184 @@
+use semver::Version;
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+type OpenAPIFileName = String;
+type OpenAPIVersionRaw = String;
+
+const TYPE_ALIAS: &str = r"/// Open API file name
+pub type OpenAPIFileName = String;
+";
+
+fn list_all_open_api_spec_files(paths: &[&Path]) -> Vec<PathBuf> {
+    let mut open_api_spec_files = Vec::new();
+
+    for path in paths {
+        for entry in crate::list_files_in_folder(path).filter(|e| {
+            let os_filename = e.file_name();
+            let filename = os_filename.to_string_lossy();
+            filename.starts_with("openapi") && filename.ends_with(".yaml")
+        }) {
+            open_api_spec_files.push(entry.path())
+        }
+    }
+
+    open_api_spec_files
+}
+
+fn read_version_from_open_api_spec_file(spec_file_path: PathBuf) -> OpenAPIVersionRaw {
+    let yaml_spec = fs::read_to_string(spec_file_path).unwrap();
+    let open_api: serde_yaml::Value = serde_yaml::from_str(&yaml_spec).unwrap();
+    open_api["info"]["version"].as_str().unwrap().to_owned()
+}
+
+/// Generate the `get_open_api_versions_mapping` function based on the openapi files
+/// in the given folders.
+pub fn generate_open_api_versions_mapping(paths: &[&Path]) -> String {
+    let open_api_spec_files = list_all_open_api_spec_files(paths);
+    // Use a BTreeMap to guarantee the order
```

Comment:
Maybe this is more explanatory as the concern is to generate always the same code?
```suggestion
    // Use a BTreeMap to guarantee the deterministic code generation below
```